includes/collection/class-outbox.php

```diff
@@ -296,28 +296,35 @@ public static function maybe_get_activity( $outbox_item ) {
 	/**
 	 * Get the object ID of an activity.
 	 *
-	 * @param Activity $activity The activity object.
+	 * @param Generic_Object $data The activity object.
+	 *
 	 * @return string The object ID.
 	 */
-	private static function get_object_id( $activity ) {
+	private static function get_object_id( $data ) {
+		// If the object is an array, convert it to a Generic_Object.
+		if ( is_array( $data->get_object() ) ) {
```

Comment:
Are there instance outside of the unit tests, where we'd expect the object to be an array?

Reply:
What happens if you use an array as param for set_object? Will it be transformed?

Reply:
Sounds like something we'd want to handle in set_object

Reply:
Nevertheless there could be arrays already in the outbox!?!

Reply:
What do you think of [this exploration](https://github.com/Automattic/wordpress-activitypub/compare/fix/outbox-object-id...suggest/object_id)?

It uses Activities and nested Activities instead of generic objects, in the hopes that that is a bit closer to what we'll see in real life?

Reply:
I will check it out on Monday, thanks for the input!

...and you challenged me once again and I had an idea how to properly cast to the correct object in the init_from_array function while having a bike-ride ☺️

Reply:
Not sure that's necessary if we call it on the right class

Reply:
But let's take the case where you want to fill the whole traverse that means the surrounding object with an activity as object with an object as object! We would have to build something that can handle that "filling" process recursively.

Hard to explain... but I also had no detailed look at your proposal...

Reply:
I think I know what you mean. In the proposal it creates an Activity and then passes it as an object to another Activity.

So far, I think, we always had an activity that we want to wrap in another one, never a multi-level activity that needed transforming?